pymongo/pool.py

```diff
@@ -1602,6 +1615,7 @@ def connect(self, handler: Optional[_MongoClientErrorHandler] = None) -> Connect
             raise
 
         conn = Connection(sock, self, self.address, conn_id)  # type: ignore[arg-type]
+        self.active_conns.append(conn)
```

Comment:
I'm a little concerned about the impact of adding another place where Connections are stored. In particular around potential bugs where `active_conns` becomes out of sync with the pool, or a connection is kept alive because of this reference. It could be safer to instead store the CancellationContexts since those have no resources and are take up less memory. What do you think?

Reply:
I like this idea, we only need to track contexts, not connections themselves.